content/blog/2017/2017-02-02-scm-api-2-take2.adoc

```diff
@@ -0,0 +1,116 @@
+---
+:layout: post
+:title: SCM API 2.0 Release Take 2
+:tags:
+- development
+- plugins
+:author: stephenc
+---
+
+In January we link:https://jenkins.io/blog/2017/01/17/scm-api-2/[announced the release of SCM API 2.0].
+There were four important issues identified after the original release was published and in order to allow those issues to be resolved, the updates were pulled from the update center.
+
+These issues were:
+
+link:https://issues.jenkins-ci.org/browse/JENKINS-41121[JENKINS-41121]::
+Migration of GitHub branches from 1.x to 2.x resulted in a change of branch identity which, now that other bugs in Branch API had been fixed, resulted in a rebuild of all GitHub Branches (not PRs) after an upgrade to GitHub Branch Source Plugin 2.0.0.
++
+*Resolution:* The SCM API plugin was enhanced to add an extension point that allows for a second round of data migration when upgrading.
+The Branch API plugin was enhanced to use this new extension point.
+The GitHub Branch Source plugin was enhanced to provide an implementation of this extension point.
+
+link:https://issues.jenkins-ci.org/browse/JENKINS-41255[JENKINS-41255]::
+The GitHub Branch Source and BitBucket Branch Source plugins in 1.x were not assigning consistent IDs to multi-branch projects discovered in an Organization Folder.
+Both plugins were fixed in 2.0.0 to assign consistent IDs as a change of ID would result in a rebuild of all projects.
+What was missed is that the very first scan of an Organization Folder after an upgrade will change the randomly assigned ID assigned by the 1.x plugins into the consistent ID assigned by the 2.0.0 plugins and consequently trigger a rebuild of all branches.
```

Comment:
trigger a rebuild of all branches, aka a build storm.

You use "build storm" in the resolution so it should be defined in the problem